### ISSUE TITLE
new-haddock fails with GHC-7.8.4 bundled haddock

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#5339]
```
setup: Unrecognised flags: lib:servant
cabal: Failed to build documentation for servant-0.13.0.1 (which is required
by tutorial-0.10).
```

I think it's because haddock is old. I'd prefer either solution:
- say to upgrade haddock and fail *very early*
- don't fail

**Comment by alexbiehl** [comment]
This is solved in https://github.com/haskell/cabal/pull/5261

**Comment by phadej** [comment]
@alexbiehl thanks for the pointer, I'll try that

**Comment by alexbiehl** [comment]
Please report on that PR if it worked for you.

**Comment by 23Skidoo** [comment]
Fixed by #5261.
